processor/transformprocessor/internal/common/expression.go

```diff
@@ -24,6 +24,7 @@ type TransformContext interface {
 	GetItem() interface{}
 	GetInstrumentationScope() pcommon.InstrumentationScope
 	GetResource() pcommon.Resource
+	GetDescriptor() interface{}
```

Comment:
We can probably cast the context instead of the return of `GetDescriptor` to avoid adding the GetDescriptor method - unlike GetItem (which also arguably isn't important for a similar reason but still a bit more useful), we know the other signals won't ever fill this in

Reply:
Updated.  Since we are not adding it at the `TransformContext` interface and the concept will only apply to metrics, I opted to change the name from `descriptor` to `metric`.  I think this matches the protobuf closer.  With `metric`, you can access fields like `metric.name` or `metric.is_monotonic` instead of `descriptor.metric_name` or `descriptor.metric_is_monotonic`. I think this will be easier for users, since the field name in the protobuf is `name` not `metric_name`. What do you think?